### PR TITLE
🌱 Simplify testing nightly builds with clusterctl

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Provide a quick solution to test nightly builds with clusterctl

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/9999

/area devtools

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@
 apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
 kind: Metadata
 releaseSeries:
```
Comment (sbueringer) [inline metadata.yaml:8]: Nice catch!

@cahillsf Usually (1.5 & 1.6) we did this as part of "Prepare main branch for development of the new release".

Looks like we missed it for v1.7, but it also looks like it is not documented in release-tasks

File: cmd/clusterctl/hack/create-local-repository.py [hunk 2]
```diff
@@ -167,6 +169,9 @@ def create_local_repositories():
     assert providerList is not None, 'invalid configuration: please define the list of providers to override'
     assert len(providerList)>0, 'invalid configuration: please define at least one provider to override'
 
+    if len(sys.argv) == 1:
+        execCmd(['make', 'kustomize'])
```
Comment (sbueringer) [inline cmd/clusterctl/hack/create-local-repository.py:173]: Why are we now additionally running make kustomize in this case?

As far as I can tell the only time we use it we run it directly before that (l.201-202)

But I'm probably missing something

Comment (fabriziopandini) [reply cmd/clusterctl/hack/create-local-repository.py:173]: before we were calling make customize several times, one for each provider, now we are calling it once, and only if we are actually going to use it later

File: cmd/clusterctl/hack/create-local-repository.py [hunk 3]
```diff
@@ -68,19 +70,19 @@
...
     'infrastructure-docker': {
-        'componentsFile': 'infrastructure-components.yaml',
+        'componentsFile': 'infrastructure-components-development.yaml',
         'nextVersion': 'v1.7.99',
         'type': 'InfrastructureProvider',
         'configFolder': 'test/infrastructure/docker/config/default',
     },
     'infrastructure-in-memory': {
-          'componentsFile': 'infrastructure-components.yaml',
+          'componentsFile': 'infrastructure-components-development.yaml',
```
Comment (sbueringer) [inline cmd/clusterctl/hack/create-local-repository.py:79]: If I see correctly this should be "artifacts.k8s-staging-cluster-api.appspot.com/components/nightly_main_20240101/infrastructure-components-in-memory-development.yaml"

File: cmd/clusterctl/hack/create-local-repository.py (hunk 2 above)
Comment (sbueringer) [reply cmd/clusterctl/hack/create-local-repository.py:173]: Should we then drop l.201?

File: cmd/clusterctl/hack/create-local-repository.py (hunk 3 above)
Comment (fabriziopandini) [reply cmd/clusterctl/hack/create-local-repository.py]: Nice catch, fixed

File: cmd/clusterctl/hack/create-local-repository.py (hunk 2 above)
Comment (sbueringer) [reply cmd/clusterctl/hack/create-local-repository.py:173]: @fabriziopandini ^^

**Comment by sbueringer** [comment]
Thank you!

/approve

/assign @chrischdi

File: metadata.yaml (hunk 1 above)
Comment (cahillsf) [reply metadata.yaml:8]: ah, it is in [the tasks](https://docs.google.com/document/d/157pNz7xpORkiMVYFyDIMJIRShWxVBxiThF20ymlWaHc/edit?pli=1) but i missed it this time.  thanks for fixing!

Comment (sbueringer) [reply metadata.yaml:8]: Okay I don't know what this google doc is :) (I don't have access)

Shouldn't it be documented in the publicly accessible release-tasks.md? (I didn't find it there, but maybe I missed it)

**Review by chrischdi — COMMENTED** [review]
/lgtm

File: metadata.yaml (hunk 1 above)
Comment (cahillsf) [reply metadata.yaml:8]: Ah I dropped the wrong link, just fixed
> Add the new release to the root level metadata.yaml

Comment (sbueringer) [reply metadata.yaml:8]: Ah :D 

Okay somehow I missed it, but it's there
> Add the new release to the root level metadata.yaml

**Pull request merged**